### PR TITLE
[DOCS-4684] Add images to Obs Pipelines Prod Deployment Overview docs

### PR DESCRIPTION
**Pull request by maycmlee**
### What does this PR do?

Adds images to Obs Pipelines' Prod Deployment Overview docs.

### Motivation

[DOCS-4684](https://datadoghq.atlassian.net/browse/DOCS-4684)

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

[DOCS-4684]: https://datadoghq.atlassian.net/browse/DOCS-4684?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by barieom — APPROVED** [review]
Looks great, thank you for updating this!

File: content/en/observability_pipelines/production_deployment_overview/architecture_design_and_principles.md [hunk 1]
```diff
@@ -94,7 +100,11 @@ Observability Pipelines Worker can be deployed anywhere in your infrastructure.
 
 #### Local processing
 
-With local processing, Observability Pipelines Worker is deployed on each node as an agent. Data is processed on the same node from which the data originated. This provides operational simplicity since the Observability Pipelines Worker has direct access to your data and scales along with your infrastructure.
+With local processing, Observability Pipelines Worker is deployed on each node as an agent. 
+
+{{< img src="observability_pipelines/production_deployment_overview/agent.png" alt="A diagram showing two separate nodes, each containing services, other agents, other node data, and an Observability Pipelines Worker, and the node and agent data is sent to the Worker, and then to the sinks" style="width:70%;" >}}
```
Comment (rtrieu) [inline content/en/observability_pipelines/production_deployment_overview/architecture_design_and_principles.md:105]:
```suggestion
{{< img src="observability_pipelines/production_deployment_overview/agent.png" alt="A diagram showing two separate nodes, each containing services, other agents, other node data, and an Observability Pipelines Worker, from which the node and agent data is sent to the Worker, and then to the sinks" style="width:70%;" >}}
```

File: content/en/observability_pipelines/production_deployment_overview/architecture_design_and_principles.md [hunk 2]
```diff
@@ -104,14 +114,20 @@ Local processing is recommended for:
...
+
+Data processing is shifted off your nodes and onto remote aggregator nodes. Remote processing is recommended for environments that require high durability and high availability (most environments). In addition, this is easier to set up since it does not require the infrastructure restructuring necessary when adding an agent.
 
 See [Aggregator Architecture][5] for more details.
 
 #### Unified processing
 
 Finally, you can also combine local and remote data processing to create a unified observability data pipeline. Datadog recommends evolving towards unified processing after starting with [remote processing](#remote-processing).
 
+{{< img src="observability_pipelines/production_deployment_overview/unified.png" alt="A diagram showing a node containing multiple services and another agent, both sending data to the Worker in the node, and the Worker sends the data to the load balancers. The load balancer then sends data to the aggregator, which containers multiple Workers" style="width:70%;" >}}
```
Comment (rtrieu) [inline content/en/observability_pipelines/production_deployment_overview/architecture_design_and_principles.md:129]:
```suggestion
{{< img src="observability_pipelines/production_deployment_overview/unified.png" alt="A diagram showing a node containing multiple services and another agent, both sending data to the Worker in the node, and the Worker sends the data to the load balancers. The load balancer then sends data to the aggregator, which contains multiple Workers" style="width:70%;" >}}
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Left a few suggestions, but otherwise lovely update!

**Review by maycmlee — COMMENTED** [review]
Thanks @rtrieu for reviewing and the feedback! Could you re-review?

**Review by rtrieu — APPROVED** [review]
👍 🎉

**Pull request merged**